sway-lib-std/src/constants.sw

```diff
@@ -1,6 +1,12 @@
 library constants;
 
-use ::contract_id::ContractId;
+use ::{
+  address::Address,
+  contract_id::ContractId,
+  identity::Identity,
+};
 
 pub const BASE_ASSET_ID = ~ContractId::from(ZERO_B256);
+pub const ZERO_ADDRESS = ~Address::from(ZERO_B256);
 pub const ZERO_B256 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+pub const ZERO_IDENTITY = ~Identity::Address(ZERO_ADDRESS);
```

Comment:
```suggestion
pub const ZERO_IDENTITY = Identity::Address(ZERO_ADDRESS);
```

Reply:
We have a zero contractid already (the base asset) so thought a zero address would also be useful.

Reply:
The zero contract ID exists because people might want to refer to the base asset, but TBH it should be renamed to `BASE_ASSET_ID`, where its value should not be relied on to be zero. The zero address serves no purpose.

Reply:
It is called `BASE_ASSET_ID` already :) We also have `ZERO_B256` but that's just a raw `b256` with no meaning.

And +1 to what John said.

Reply:
Okay so should I remove the zero address and instead use the base_asset_id in the identity? The problem I see is that the zero contract id actually refers to something, that being ETH, whereas the 0 address doesnt mean anything, making it more suitable for a "zero identity"

Reply:
Would `Option` not solve your problem in this case?

Reply:
Identity is a wrapper of 2 variants. Selecting one of the variants is a bad idea.

Reply:
Yeah relying on default values is a source of vulnerabilities. `Option` is the way to go here I think even if it's a bit more expensive. Do you agree? And if so, can this PR be closed?